src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,97 @@
+package duke;
+
+import exceptions.BadInputException;
+
+/**
+ * Represents a parser to parse the inputs from the user.
+ */
+public class Parser {
+    private final String command;
+    private final String arguments;
+
+    /**
+     * Constructs a parser to parse the inputs from the user.
+     *
+     * @param input
+     *            the input from the user
+     * @throws BadInputException
+     *             if the input is empty
+     */
+    public Parser(String input) throws BadInputException {
+        String[] inputParts = input.split(" ", 2);
+        if (inputParts.length < 1) {
+            throw new BadInputException(
+                    "Please enter a command!",
+                    "list, mark, unmark, todo, deadline, event, delete, bye",
+                    null,
+                    null);
+        }
+        this.command = inputParts[0];
+        this.arguments = inputParts.length > 1 ? inputParts[1] : null;
```

Comment:
The condition in the ternary expression should be bracketed.
```suggestion
        this.arguments = (inputParts.length > 1) ? inputParts[1] : null;
```